pkg/volume/gce_pd/attacher.go

```diff
@@ -96,7 +96,21 @@ func (attacher *gcePersistentDiskAttacher) Attach(spec *volume.Spec, nodeName ty
 	return path.Join(diskByIdPath, diskGooglePrefix+pdName), nil
 }
 
-func (attacher *gcePersistentDiskAttacher) VolumesAreAttached(specs []*volume.Spec, nodeName types.NodeName) (map[*volume.Spec]bool, error) {
+func (attacher *gcePersistentDiskAttacher) VolumesAreAttached(volumesByNode map[types.NodeName][]*volume.Spec) (map[*volume.Spec]bool, error) {
+	volumesAttachedCheck := make(map[*volume.Spec]bool)
+	for nodeName, specs := range volumesByNode {
+		nodeVolumesAttachedCheck, err := attacher.volumesAreAttachedToNode(specs, nodeName)
```

Comment:
Comment by @jsafrane reminds me - now that we are doing this serially for cloudproviders that don't do bulk volume polling, if user is running a 1000 node cluster and assuming each API calls takes `1 second to` request/response/parse - this function will run for 17 minutes. That may be bit extreme. cc @jingxu97